commit e5672bfe1606d1ad3c8db3251bbc914ad9e609d5
Author: Panu Matilainen <pmatilai@redhat.com>
Date:   Fri Sep 12 20:34:04 2014 +0300

    Dependency refactor whack-a-mole, aka One More Time...
    
    - Commit 0bda2faa4de368a87f85084856a5fed701774acb fixed manual
      dependencies in rpmspec queryies, but missed automatic self-provides.
      Add the self-provides before the first round of header dependency
      population to fix. SIGH.
    - Another sorta related side-effect is that the exact order of rpmspec
      output changes as things are now properly sorted, previously it
      was a mixed bag.
    
    (cherry picked from commit 4bc2900baee9ff289dabc47609059accc0f39983)

diff --git a/build/parseSpec.c b/build/parseSpec.c
index 77cf409..ea761c9 100644
--- a/build/parseSpec.c
+++ b/build/parseSpec.c
@@ -549,12 +549,12 @@ static void addTargets(Package Pkgs)
 	headerPutString(pkg->header, RPMTAG_OPTFLAGS, optflags);
 
 	/* Add manual dependencies early for rpmspec etc to look at */
+	addPackageProvides(pkg);
 	for (int i=0; i<PACKAGE_NUM_DEPS; i++) {
 	    rpmdsPutToHeader(pkg->dependencies[i], pkg->header);
 	}
 
 	pkg->ds = rpmdsThis(pkg->header, RPMTAG_REQUIRENAME, RPMSENSE_EQUAL);
-	addPackageProvides(pkg);
     }
     free(platform);
     free(arch);
